src/NHibernate/Dialect/PostgreSQL81Dialect.cs

```diff
@@ -34,6 +33,13 @@ namespace NHibernate.Dialect
 	/// <seealso cref="PostgreSQLDialect" />
 	public class PostgreSQL81Dialect : PostgreSQLDialect
 	{
+		protected override void RegisterDateTimeTypeMappings()
+		{
+			base.RegisterDateTimeTypeMappings();
+			RegisterColumnType(DbType.DateTime, 6, "timestamp($s)");
```

Comment:
What will be the resolution of `$s=7`?

Reply:
Not handled by postgresql.

Reply:
But will it resolve to just `timestamp`?

Reply:
No idea, missing test.

Reply:
Still sent in creation script as `timestamp(7)`, PostgreSql switches it by itself to `timestamp(6)`. Other db may yield an error instead.

The 6 in `RegisterColumnType(DbType.DateTime, 6, "timestamp($s)")` in facts works only for size/length, it has no effect on scale. There are no mechanism for this currently.

It could be easily adapted by considering [here](https://github.com/nhibernate/nhibernate-core/blob/master/src/NHibernate/Dialect/TypeNames.cs#L86) that for `DbType.DateTime/DateTime2`, the size does not matter and we should consider the scale. The current implementation would then yield `timestamp` for column creation. Db compliant to the SQL standard should then yield 6 (according to [MySql doc](https://dev.mysql.com/doc/refman/5.7/en/fractional-seconds.html)). But MySql would yield 0, Sql Server 7, Postgres writes "no explicit bound" (what does it mean? May be more than 6 in such case?) and Oracle 6.

So if adapting the code in `TypeNames`, maybe should we default to explicit max scale instead of the bare type without scale.

Reply:
> The current implementation would then yield timestamp for column creation.

That what is doing Postgres dialect because I have forgotten to override default type with `timestamp(6)` instead. So no needs for complicating things, just need to handle the scale types case, yielding default if above scale, and taking care of having default suitably defined to max scale in dialect.

Reply:
Done.